src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -438,9 +447,18 @@ synchronized private boolean connectOne(long sid, InetSocketAddress electionAddr
         Socket sock = null;
         try {
              LOG.debug("Opening channel to server " + sid);
-             sock = new Socket();
-             setSockOpts(sock);
-             sock.connect(electionAddr, cnxTO);
+             if (self.isSslQuorum()) {
+                 SSLSocket sslSock = x509Util.createSSLSocket();
+                 setSockOpts(sslSock);
+                 sslSock.connect(electionAddr, cnxTO);
+                 sslSock.startHandshake();
```

Comment:
So I believe in 3.4 we have a workaround for this for SASL https://github.com/apache/zookeeper/blob/branch-3.4/src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java#L583

I agree that this is something we should document and we should fix the root cause elsewhere.